docs/writing-stories/mocking-network-requests.md

```diff
@@ -0,0 +1,180 @@
+---
+title: Mocking network requests
+---
+
+For components that make network requests (e.g. fetching data from a REST or GraphQL API), you can mock those requests using a tool like [Mock Service Worker (MSW)](https://mswjs.io/). MSW is an API mocking library, which relies on service workers to capture network requests and provides mocked data in response.
+
+The [MSW addon](https://storybook.js.org/addons/msw-storybook-addon/) brings this functionality into Storybook, allowing you to mock API requests in your stories. Below is an overview of how to set up and use the addon.
+
+## Set up the MSW addon
+
+First, if necessary, run this command to install MSW:
+
+<!-- prettier-ignore-start -->
+
+<CodeSnippets
+  paths={[
+   'common/msw-install.npm.js.mdx',
+   'common/msw-install.yarn.js.mdx',
+   'common/msw-install.pnpm.js.mdx',
+  ]}
+/>
+
+<!-- prettier-ignore-end -->
+
+Then generate the service worker file necessary for MSW to work:
+
+<!-- prettier-ignore-start -->
+
+<CodeSnippets
+  paths={[
+   'common/msw-generate-service-worker.npx.js.mdx',
+   'common/msw-generate-service-worker.pnpm.js.mdx',
+  ]}
+/>
+
+<!-- prettier-ignore-end -->
+
+<If renderer="angular">
+
+<Callout variant="info" icon="💡">
+
+Angular projects will likely need to adjust the command to save the mock service worker file in a different directory (e.g., `src`).
+
+</Callout>
+
+</If>
+
+Next, install and register the MSW addon:
+
+<!-- TODO: Snippetize -->
+
+```sh
+npx storybook@latest add msw-storybook-addon
+```
```

Comment:
To add some context, this type of addon will not require to be registered in main.js based on its architecture and as it does not provide a UI element. More even, as this one is affecting the preview. Based on some conversations, this will remain as currently documented. Potentially, in the near future, it could be optimized to:
- Be installed via CLI without it being registered
- Automatically inject the required config automatically 🤞

Reply:
Yup the addon is not really an addon in a way, depends on how you see it. It's basically a package that exports a storybook loader. It does not have a preset, it does not have a register file so adding it main.js doesn't really make sense! In the future there are plans to make it into a full on addon with automatic loader, a panel etc, but that's not the case today.